### PR TITLE
Lodash string

### PR DESCRIPTION
**Pull request by daksamit**
template test to check

### CODE REVIEW

File: test/unit/lodash.spec.js [hunk 1]
```diff
@@ -0,0 +1,568 @@
+'use strict';
+
+var chai = require('chai');
+var expect = chai.expect;
+var datasets = require('../../app/datasets');
+
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:7]: Missing "var _ = require('lodash');"

File: test/unit/lodash.spec.js [hunk 2]
```diff
@@ -0,0 +1,568 @@
...
+describe('Lodash training', function () {
+    var params;
+
+    describe('camelCase', function () {
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:11]: What if text I enter is already in camel case...

File: test/unit/lodash.spec.js [hunk 3]
```diff
@@ -0,0 +1,568 @@
...
+        before(function () {
+            params = datasets.camelCase();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+            expect(params).to.have.length(1);
+        });
+        it('should convert string to camel case', function () {
+                expect(_.camelCase.apply(_, params)).to.eql('someStringToTestFunction');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:24]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 4]
```diff
@@ -0,0 +1,568 @@
...
+            }
+        );
+    });
+
+    describe('capitalize', function () {
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:29]: should check if first sign is already uppercase

File: test/unit/lodash.spec.js [hunk 5]
```diff
@@ -0,0 +1,568 @@
...
+        before(function () {
+            params = datasets.capitalize();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+            expect(params).to.have.length(1);
+        });
+        it('should convert the first character of string to upper case and the remaining to lower case', function () {
+                expect(_.capitalize.apply(_, params)).to.eql('Some string to test function');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:41]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 6]
```diff
@@ -0,0 +1,568 @@
...
+
+    describe('deburr', function () {
+        before(function () {
+            params = datasets.deburr();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+            expect(params).to.have.length(1);
+        });
+        it('should remove combining diacritical marks from string', function () {
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:57]: If this should remove extra marks, it should be some extra marks in text before...

File: test/unit/lodash.spec.js [hunk 7]
```diff
@@ -0,0 +1,568 @@
...
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+            expect(params[1]).to.be.a('string');
+            expect(params).to.have.length.within(2, 3);
+            if (params[2] !== undefined && params.length === 3) {
+                expect(Number.isInteger(params[2])).to.eql(true);
+            }
+        });
+        it('should check if letter on given position is a target string', function () {
+                expect(params[0][params[2] - 1]).to.eql(params[1]);
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js]: In my solution "['abc', 'c']" `params[0][params[2] - 1]` returns `undefined`

File: test/unit/lodash.spec.js [hunk 8]
```diff
@@ -0,0 +1,568 @@
...
+    describe('deburr', function () {
+        before(function () {
+            params = datasets.deburr();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+            expect(params).to.have.length(1);
+        });
+        it('should remove combining diacritical marks from string', function () {
+                expect(_.deburr.apply(_, params)).to.eql('some string to test function');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:58]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 9]
```diff
@@ -0,0 +1,568 @@
...
+    describe('escape', function () {
+        before(function () {
+            params = datasets.escape();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+            expect(params).to.have.length(1);
+        });
+        it('should return converted string', function () {
+                expect(_.escape.apply(_, params)).to.eql('Tom &amp; Jerry are old friends');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:96]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 10]
```diff
@@ -0,0 +1,568 @@
...
+    describe('escapeRegExp', function () {
+        before(function () {
+            params = datasets.escapeRegExp();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+            expect(params).to.have.length(1);
+        });
+        it('should return converted string', function () {
+                expect(_.escapeRegExp.apply(_, params)).to.eql('\\[realskill\\]\\(https://realskill\\.com/\\)');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:110]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 11]
```diff
@@ -0,0 +1,568 @@
...
+
+    describe('kebabCase', function () {
+        before(function () {
+            params = datasets.kebabCase();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+            expect(params).to.have.length(1);
+        });
+        it('shold return transformed string', function () {
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:127]: It should check if string is already in kebabCase

File: test/unit/lodash.spec.js [hunk 12]
```diff
@@ -0,0 +1,568 @@
...
+                expect(_.kebabCase.apply(_, params)).to.eql('some-string-to-test-function');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:128]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 13]
```diff
@@ -0,0 +1,568 @@
...
+
+    describe('lowerCase', function () {
+        before(function () {
+            params = datasets.lowerCase();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+            expect(params).to.have.length(1);
+        });
+        it('should return string converted to lowercase', function () {
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js]: should check in advance if there is some big letter in text

File: test/unit/lodash.spec.js [hunk 14]
```diff
@@ -0,0 +1,568 @@
...
+                expect(_.lowerCase.apply(_, params)).to.eql('some string to test function');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:150]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 15]
```diff
@@ -0,0 +1,568 @@
...
+
+    describe('lowerFirst', function () {
+        before(function () {
+            params = datasets.lowerFirst();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+            expect(params).to.have.length(1);
+        });
+        it('should return converted string with first letter in lowercase', function () {
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js]: What if first letter is already in lowercase?

File: test/unit/lodash.spec.js [hunk 16]
```diff
@@ -0,0 +1,568 @@
...
+                expect(_.lowerFirst.apply(_, params)).to.eql('sOME STRING TO TEST FUNCTION');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:168]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 17]
```diff
@@ -0,0 +1,568 @@
...
+    describe('pad', function () {
+        before(function () {
+            params = datasets.pad();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+            expect(Number.isInteger(params[1])).to.eql(true);
+            expect(params[1]).to.be.above(params[0].length);
+            expect(params).to.have.length.within(2, 3);
+            if (params[2] !== undefined && params.length === 3) {
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js]: Solution showed underneath required third parameter

File: test/unit/lodash.spec.js [hunk 18]
```diff
@@ -0,0 +1,568 @@
...
+                expect(params[2]).to.be.a('string');
+            }
+        });
+        it('should return converted string with padding characters', function () {
+                expect(_.pad.apply(_, params)).to.eql('._._.some long sweet text._._.');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:189]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 19]
```diff
@@ -0,0 +1,568 @@
...
+    describe('padEnd', function () {
+        before(function () {
+            params = datasets.padEnd();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+            expect(Number.isInteger(params[1])).to.eql(true);
+            expect(params[1]).to.be.above(params[0].length);
+            expect(params).to.have.length.within(2, 3);
+            if (params[2] !== undefined && params.length === 3) {
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js]: Solution showed underneath required third parameter

File: test/unit/lodash.spec.js [hunk 20]
```diff
@@ -0,0 +1,568 @@
...
+                expect(params[2]).to.be.a('string');
+            }
+        });
+        it('should return converted string with padding characters on right', function () {
+                expect(_.padEnd.apply(_, params)).to.eql('some another long sweet text.*.*.*.*');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:210]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 21]
```diff
@@ -0,0 +1,568 @@
...
+    describe('padStart', function () {
+        before(function () {
+            params = datasets.padStart();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+            expect(Number.isInteger(params[1])).to.eql(true);
+            expect(params[1]).to.be.above(params[0].length);
+            expect(params).to.have.length.within(2, 3);
+            if (params[2] !== undefined && params.length === 3) {
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js]: Solution showed underneath required third parameter

File: test/unit/lodash.spec.js [hunk 22]
```diff
@@ -0,0 +1,568 @@
...
+                expect(params[2]).to.be.a('string');
+                expect(params[2]).to.have.length.above(0);
+            }
+        });
+        it('should return converted string with padding characters on left', function () {
+                expect(_.padStart.apply(_, params)).to.eql('|_|_|_|_|_|_try again some long sweet text');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:232]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 23]
```diff
@@ -0,0 +1,568 @@
...
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+            expect(params).to.have.length.within(1, 2);
+            if (params[1] !== undefined && params.length === 2) {
+                expect(Number.isInteger(params[1])).to.eql(true);
+            }
+        });
+        it('should return converted string to integer', function () {
+                expect(_.parseInt.apply(_, params)).to.eql(8);
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:253]: Why 8? Why not 7 or 88...

File: test/unit/lodash.spec.js [hunk 24]
```diff
@@ -0,0 +1,568 @@
...
+    describe('repeat', function () {
+        before(function () {
+            params = datasets.repeat();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+            expect(Number.isInteger(params[1])).to.eql(true);
+        });
+        it('should return new repeated string', function () {
+                expect(_.repeat.apply(_, params)).to.eql('repeat me!repeat me!repeat me!repeat me!');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:271]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 25]
```diff
@@ -0,0 +1,568 @@
...
+            before(function () {
+                params = datasets.replace1();
+            });
+            it('should match types of passing elements', function () {
+                expect(params[0]).to.be.a('string');
+                expect(params[1]).to.be.a('string');
+                expect(params[2]).to.be.a('string');
+            });
+            it('should return modified string', function () {
+                    expect(_.replace.apply(_, params)).to.eql('Hello universe!');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:287]: Why "Hello universe!"?

File: test/unit/lodash.spec.js [hunk 26]
```diff
@@ -0,0 +1,568 @@
...
+            before(function () {
+                params = datasets.replace2();
+            });
+            it('should match types of passing elements', function () {
+                expect(params[0]).to.be.a('string');
+                expect(params[1]).to.be.a('RegExp');
+                expect(params[2]).to.be.a('string');
+            });
+            it('should return modified string', function () {
+                    expect(_.replace.apply(_, params)).to.eql('Hello world!');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:301]: Why "Hello world!"?

File: test/unit/lodash.spec.js [hunk 27]
```diff
@@ -0,0 +1,568 @@
...
+            before(function () {
+                params = datasets.replace3();
+            });
+            it('should match types of passing elements', function () {
+                expect(params[0]).to.be.a('string');
+                expect(params[1]).to.be.a('string');
+                expect(params[2]).to.be.a('function');
+            });
+            it('should return modified string', function () {
+                    expect(_.replace.apply(_, params)).to.eql('Hello world!');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:315]: Why "Hello world!"?

File: test/unit/lodash.spec.js [hunk 28]
```diff
@@ -0,0 +1,568 @@
...
+            before(function () {
+                params = datasets.replace4();
+            });
+            it('should match types of passing elements', function () {
+                expect(params[0]).to.be.a('string');
+                expect(params[1]).to.be.a('RegExp');
+                expect(params[2]).to.be.a('function');
+            });
+            it('should return modified string', function () {
+                    expect(_.replace.apply(_, params)).to.eql('Hello world!');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:329]: Why?

File: test/unit/lodash.spec.js [hunk 29]
```diff
@@ -0,0 +1,568 @@
...
+                }
+            );
+        });
+    });
+
+    describe('snakeCase', function () {
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:335]: Should check if text is not in snakeCase already

File: test/unit/lodash.spec.js [hunk 30]
```diff
@@ -0,0 +1,568 @@
...
+        before(function () {
+            params = datasets.snakeCase();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+        });
+        it('should return converted string', function () {
+                expect(_.snakeCase.apply(_, params)).to.eql('some_string_to_test_function');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:347]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 31]
```diff
@@ -0,0 +1,568 @@
...
+        before(function () {
+            params = datasets.split();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+            expect(params[1]).to.be.a('string');
+            expect(Number.isInteger(params[2])).to.eql(true);
+        });
+        it('should return the string segments', function () {
+                expect(_.split.apply(_, params)).to.eql(['split', 'me']);
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:362]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 32]
```diff
@@ -0,0 +1,568 @@
...
+            }
+        );
+    });
+
+    describe('startCase', function () {
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:367]: should check if text is already in startCase

File: test/unit/lodash.spec.js [hunk 33]
```diff
@@ -0,0 +1,568 @@
...
+        before(function () {
+            params = datasets.startCase();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+        });
+        it('should return converted string', function () {
+                expect(_.startCase.apply(_, params)).to.eql('Some String To Test Function');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:379]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 34]
```diff
@@ -0,0 +1,568 @@
...
+
+    describe('startsWith', function () {
+        before(function () {
+            params = datasets.startsWith();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+            expect(params[1]).to.be.a('string');
+            expect(params).to.have.length.within(2, 3);
+            if (params[2] !== undefined && params.length === 3) {
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js]: it underneath required third parameter

File: test/unit/lodash.spec.js [hunk 35]
```diff
@@ -0,0 +1,568 @@
...
+                expect(params[0][params[2]]).to.eql(params[1]);
+            }
+        );
+        it('should check if string starts with the given target string', function () {
+                expect(_.startsWith.apply(_, params)).to.eql(true);
+            }
+        );
+    });
+
+    describe('template', function () {
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:405]: I don't know how to solve that, even with documentation. Maybe some tips in readme...

File: test/unit/lodash.spec.js [hunk 36]
```diff
@@ -0,0 +1,568 @@
...
+
+    describe('toLower', function () {
+        before(function () {
+            params = datasets.toLower();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+        });
+        it('should return converted string', function () {
+                expect(_.toLower.apply(_, params)).to.eql('some string to test function');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:464]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 37]
```diff
@@ -0,0 +1,568 @@
...
+
+    describe('toUpper', function () {
+        before(function () {
+            params = datasets.toUpper();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+        });
+        it('should return converted string', function () {
+                expect(_.toUpper.apply(_, params)).to.eql('SOME STRING TO TEST FUNCTION');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:477]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 38]
```diff
@@ -0,0 +1,568 @@
...
+            }
+        );
+    });
+
+    describe('trim', function () {
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:482]: Can be solve with empty ('') second parameter

File: test/unit/lodash.spec.js [hunk 39]
```diff
@@ -0,0 +1,568 @@
...
+            expect(params[0]).to.be.a('string');
+            expect(params[1]).to.be.a('string');
+        });
+        it('should return trimmed string', function () {
+                expect(_.trim.apply(_, params)).to.eql('some string to test function');
+            }
+        );
+    });
+
+    describe('trimEnd', function () {
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:497]: Can be solve with empty ('') second parameter

File: test/unit/lodash.spec.js [hunk 40]
```diff
@@ -0,0 +1,568 @@
...
+            expect(params[0]).to.be.a('string');
+            expect(params[1]).to.be.a('string');
+        });
+        it('should return trimmed string', function () {
+                expect(_.trimEnd.apply(_, params)).to.eql('some string to test');
+            }
+        );
+    });
+
+    describe('trimStart', function () {
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:512]: Can be solved with empty ('') second parameter

File: test/unit/lodash.spec.js [hunk 41]
```diff
@@ -0,0 +1,568 @@
...
+            if (params[1] !== undefined && params.length === 2) {
+                expect(params[1]).to.be.an('object');
+            }
+        });
+        it('should check if second argument have validate properties', function () {
+            var options = params[1];
+            if (options !== undefined && params.length === 2) {
+                expect(options).to.have.any.keys('length', 'omission', 'separator');
+            }
+            // powinien miec max 3 wlasciowsci
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js]: ?

File: test/unit/lodash.spec.js [hunk 42]
```diff
@@ -0,0 +1,568 @@
...
+            // pierwsza number
+        });
+        it('should return truncated string', function () {
+                expect(_.truncate.apply(_, params)).to.eql('some very long text...');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:552]: Can be solved without any options, simply copy and past expected text

File: test/unit/lodash.spec.js [hunk 43]
```diff
@@ -0,0 +1,568 @@
...
+
+    describe('unescape', function () {
+        before(function () {
+            params = datasets.unescape();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+        });
+        it('should return transformed string', function () {
+                expect(_.unescape.apply(_, params)).to.eql('tom & jerry were old friends');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:565]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 44]
```diff
@@ -0,0 +1,568 @@
...
+
+    describe('upperCase', function () {
+        before(function () {
+            params = datasets.upperCase();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+        });
+        it('should return transformed string', function () {
+                expect(_.upperCase.apply(_, params)).to.eql('SOME STRING TO TEST FUNCTION');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:578]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 45]
```diff
@@ -0,0 +1,568 @@
...
+            }
+        );
+    });
+
+    describe('upperFirst', function () {
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:583]: Should check if first letter is already big

File: test/unit/lodash.spec.js [hunk 46]
```diff
@@ -0,0 +1,568 @@
...
+        before(function () {
+            params = datasets.upperFirst();
+        });
+        it('should match types of passing elements', function () {
+            expect(params[0]).to.be.a('string');
+        });
+        it('should return transformed string', function () {
+                expect(_.upperFirst.apply(_, params)).to.eql('Some string to test function');
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:595]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 47]
```diff
@@ -0,0 +1,568 @@
...
+            });
+            it('should match types of passing elements', function () {
+                expect(params[0]).to.be.a('string');
+                expect(params).to.have.length.within(1, 2);
+                if (params[1] !== undefined && params.length === 2) {
+                    expect(params[1]).to.be.a('string');
+                }
+            });
+            it('should return words of string', function () {
+                    expect(_.words.apply(_, params)).to.eql([ 'bread', 'butter', 'milk' ]);
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 48]
```diff
@@ -0,0 +1,568 @@
...
+            });
+            it('should match types of passing elements', function () {
+                expect(params[0]).to.be.a('string');
+                expect(params).to.have.length.within(1, 2);
+                if (params[1] !== undefined && params.length === 2) {
+                    expect(params[1]).to.be.a('RegExp');
+                }
+            });
+            it('should return words of string', function () {
+                    expect(_.words.apply(_, params)).to.eql(['bread', 'butter', 'milk']);
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:629]: In my opinion, without looking to this file, that solution is impossible to reveal

File: test/unit/lodash.spec.js [hunk 49]
```diff
@@ -0,0 +1,568 @@
...
+                if (params[1] !== undefined && params.length === 2) {
+                    expect(params[1]).to.be.a('string');
+                }
+            });
+            it('should return words of string', function () {
+                    expect(_.words.apply(_, params)).to.eql([ 'bread', 'butter', 'milk' ]);
+                }
+            );
+        });
+        describe('words2', function () {
```
Comment (Nigatsu) [inline test/unit/lodash.spec.js:617]: Both I can solve without RegExp parameter. And what is the difference between words1 and words2?

File: app/datasets.js [hunk 50]
```diff
@@ -0,0 +1,107 @@
...
+    },
+    upperFirst: function () {
+        return [];
+    },
+    words1: function () {
+        return [];
+    },
+    words2: function () {
+        return [];
+    },
```
Comment (Nigatsu) [inline app/datasets.js:106]: unnecessary coma

File: test/unit/lodash.spec.js [hunk 51]
```diff
@@ -0,0 +1,634 @@
...
+    describe('template', function () {
+        beforeEach(function () {
+            params = datasets.template();
+        });
+        it('should match types of passing elements', function () {
+            var param1 = params[0];
+            var param2 = params[1];
+            expect(param1).to.be.a('string');
+            expect(params).to.have.length.within(1, 2);
+            if (param2 !== undefined && params.length == 2) {
```
Comment (striok) [inline test/unit/lodash.spec.js]: It should be === not ==

File: test/unit/lodash.spec.js [hunk 52]
```diff
@@ -0,0 +1,634 @@
...
+                expect(param2).to.be.an('object');
+                expect(Object.keys(param2)).to.have.length.within(1, 6);
+            }
+        });
+        it('should contain specific option parameter', function () {
+            var param2 = params[1];
+            var options = ['escape', 'evaluate', 'imports', 'interpolate', 'sourceURL', 'variable'];
+
+            expect(param2).to.have.any.keys(options);
+            for (var key in param2) {
```
Comment (striok) [inline test/unit/lodash.spec.js:424]: You should check key in this loop by if(key.hasOwnProperty(param2[key])){ ... }

File: test/unit/lodash.spec.js [hunk 53]
```diff
@@ -0,0 +1,634 @@
...
+                    expect(key).to.be.a('RegExp');
+                }
+                if (key === options[4]) {
+                    expect(key).to.be.a('string');
+                }
+            }
+        });
+        it('should not contain empty option parameter', function () {
+                var param2 = params[1];
+                for (var key in param2) {
```
Comment (striok) [inline test/unit/lodash.spec.js:447]: You should check key in this loop by if(key.hasOwnProperty(param2[key])){ ... }

File: test/unit/lodash.spec.js (hunk 17 above)
Comment (striok) [reply test/unit/lodash.spec.js]: Solution still showed underneath required third parameter

File: test/unit/lodash.spec.js (hunk 2 above)
Comment (striok) [reply test/unit/lodash.spec.js:11]: Fix

File: test/unit/lodash.spec.js (hunk 19 above)
Comment (striok) [reply test/unit/lodash.spec.js]: Solution still showed underneath required third parameter

File: test/unit/lodash.spec.js (hunk 21 above)
Comment (striok) [reply test/unit/lodash.spec.js]: Solution still showed underneath required third parameter

File: test/unit/lodash.spec.js (hunk 29 above)
Comment (striok) [reply test/unit/lodash.spec.js:335]: fix

File: test/unit/lodash.spec.js (hunk 32 above)
Comment (striok) [reply test/unit/lodash.spec.js:367]: fix

File: test/unit/lodash.spec.js (hunk 34 above)
Comment (striok) [reply test/unit/lodash.spec.js]: Solution still showed underneath required third parameter

File: test/unit/lodash.spec.js (hunk 35 above)
Comment (striok) [reply test/unit/lodash.spec.js:405]: or add some comment in datasets.js how to solve it.

File: test/unit/lodash.spec.js (hunk 38 above)
Comment (striok) [reply test/unit/lodash.spec.js:482]: fix

File: test/unit/lodash.spec.js (hunk 39 above)
Comment (striok) [reply test/unit/lodash.spec.js:497]: fix

File: test/unit/lodash.spec.js (hunk 40 above)
Comment (striok) [reply test/unit/lodash.spec.js:512]: fix

File: test/unit/lodash.spec.js (hunk 45 above)
Comment (striok) [reply test/unit/lodash.spec.js:583]: fix

File: test/unit/lodash.spec.js (hunk 4 above)
Comment (striok) [reply test/unit/lodash.spec.js:29]: fix

**Pull request closed without merging**